File: Programmers/모든_문제/연습문제/Solution_Level_1_제일작은수제거하기.py
def solution(arr):
    if len(arr) <= 1:
        return [-1]
    else:
        min_val = min(arr)
        arr.remove(min_val)
    return arr
    
def solution1(arr): 
    arr.remove(min(arr))
    return [-1] if arr == [] else arr
    
def solution2(arr):   
    return [i for i in arr if i > min(arr)] if len(arr) > 1 else [-1]

if __name__ == "__main__":
    arr = [4,3,2,1]    
    result = [4,3,2]
    print(result, solution(arr), solution1(arr), solution2(arr))

    arr, result = [10], [-1]
    print(result, solution(arr), solution1(arr), solution2(arr))
    

File: Programmers/모든_문제/연습문제/Solution_Level_1_제일작은수제거하기.java
import java.util.ArrayList;
import java.util.Arrays;

class Solution_Level_1_제일작은수제거하기 {
	
	public static void main(String[] args) {
		// 가장 작은 수를 제거한 배열을 리턴
		System.out.println(Arrays.toString(solution(new int[] {4,3,2,1})));
		System.out.println(Arrays.toString(solution(new int[] {10})));
	}
	
	
	public static int[] solution(int[] arr) {
		int min = Integer.MAX_VALUE;
		int minIdx = 0;
		for (int i = 0; i < arr.length; i++) {
			if(arr[i] < min) {
				min = arr[i];
				minIdx = i;
			}
		}
		ArrayList<Integer> li = new ArrayList<>();
		for (int i = 0; i < arr.length; i++) {
			if(i!=minIdx) {
				li.add(arr[i]);
			}
		}
		if(li.size() == 0) {
			return new int[] {-1};
		}
		return li.stream().mapToInt(x->x).toArray();
	}
	
}